src/NServiceBus.Testing.Tests/Saga/SagaDataTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace NServiceBus.Testing.Tests.Saga
+{
+    using System;
+    using System.Threading.Tasks;
+    using NUnit.Framework;
+
+    [TestFixture]
+    class SagaDataTests
+    {
+        [Test]
+        public void ShouldCreateSagaWithSagaData()
+        {
+            Test.Saga<SagaCreation>(new SagaCreation.CreationData { Property = "property" })
+                .ExpectSagaData<SagaCreation.CreationData>(data => data.Property == "property")
+                .WhenHandling<SagaCreation.NoChangeMessage>();
+        }
+
+        [Test]
+        public void ShouldUpdateSagaDataWhenChangeMessageIsCalled()
+        {
+            Test.Saga<SagaCreation>(new SagaCreation.CreationData { Property = "property" })
+                .ExpectSagaData<SagaCreation.CreationData>(data => data.Property == "newData")
+                .WhenHandling<SagaCreation.ChangeMessage>();
+        }
+    }
```

Comment:
It may make sense to have a test to verify an exception is thrown when the expectation isn't met?